### ISSUE TITLE
Syntactic sugar for "set if not empty"

### ISSUE THREAD

**Issue by msullivan** [edgedb/edgedb#4803]
It might be useful to have a variant of `:=` for inside `update ... set` to mean "set if not empty". Elvis suggested `?:=` on discord.

So `set { foo ?:= expr }` would be equivalent to `set { foo := expr ?? .foo }`.

Would this come up enough to be useful without being confusing? Or is `??` good enough?

**Comment by elprans** [comment]
`insert` can benefit from this too, `?:=` would mean "if empty, us default".

**Comment by msullivan** [comment]
So maybe this would subsume #4595. (Thought that is more flexible, but mostly not in a useful way?)

**Comment by elprans** [comment]
Yeah, this should cover enough of use cases to make #4595 unnecessary,
